Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1702,73 +1662,112 @@ extension SourceKitServer {
     return try await languageService.declaration(req)
   }
 
-  func definition(
+  /// Returns the result of a `DefinitionRequest` by running a `SymbolInfoRequest`, inspecting
+  /// its result and doing index lookups, if necessary.
+  ///
+  /// In contrast to `definition`, this does not fall back to sending a `DefinitionRequest` to the
+  /// toolchain language server.
+  private func indexBasedDefinition(
     _ req: DefinitionRequest,
     workspace: Workspace,
     languageService: ToolchainLanguageServer
-  ) async throws -> LocationsOrLocationLinksResponse? {
+  ) async throws -> [Location] {
     let symbols = try await languageService.symbolInfo(
       SymbolInfoRequest(
         textDocument: req.textDocument,
         position: req.position
       )
     )
-    let index = await self.workspaceForDocument(uri: req.textDocument.uri)?.index
+    guard let symbol = symbols.first else {
+      return []
+    }
```

Comment:
I realize the old code only handled a single result, but is there any reason not to just handle multiple for all of these?

Reply:
There is not. I thought we had an issue for this but looks like we don’t. Filed rdar://119163908